tests/src/baseservices/exceptions/WindowsEventLog/WindowsEventLog.cs

```diff
@@ -0,0 +1,137 @@
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+
+class Program
+{
+    private static void UnhandledException()
+    {
+        A();
+    }
+
+    private static void FailFast()
+    {
+        try
+        {
+            A();
+        }
+        catch (ArgumentException ex)
+        {
+            Environment.FailFast("failing fast", ex);
+        }
+
+    }
+
+    private static void A()
+    {
+        B();
+    }
+
+    private static void B()
+    {
+        throw new ArgumentException("my ae");
+    }
+
+    private static bool LaunchTest(string testName, string[] logEntriesToCheck)
+    {
+        DateTime dt = DateTime.Now;
+
+        Process testProcess = new Process();
+        string currentPath = Directory.GetCurrentDirectory();
+        string corerunPath = Environment.GetEnvironmentVariable("CORE_ROOT") + "\\corerun.exe";
+
+        testProcess.StartInfo.FileName = corerunPath;
+        testProcess.StartInfo.Arguments = currentPath + "\\WindowsEventLog.exe " + testName;
+        testProcess.StartInfo.EnvironmentVariables["CORE_ROOT"] = Environment.GetEnvironmentVariable("CORE_ROOT");
+        testProcess.StartInfo.UseShellExecute = false;
+
+        testProcess.Start();
+        testProcess.WaitForExit();
+
+        Thread.Sleep(2000);
+
+        EventLog log = new EventLog("Application");
+
+        Console.WriteLine("Found {0} entries in Event Log", log.Entries.Count);
+
+        foreach (EventLogEntry entry in log.Entries)
+        {
+            int checkCount = 0;
+            if (entry.TimeGenerated > dt.AddMinutes(-3))  // Grant some leeway in the time error was logged
```

Comment:
I'd suggest commenting more about why you needed to add leeway. Its going to look fishy to someone that doesn't understand it.

Given that the leeway is so large I also think we should do a little more to ensure that the log entries you are reading came from this invocation of the test and not some other invocation. Imagine that I have both a good and a bad coreclr installed, first I test the good one (my baseline), then a minute later I test the bad one (my new change). As long as I do the test within 3 minutes the bad one could still pass without logging anything. Some solutions that might work:
a) Count the number of log entries that match before and after the failing process. Make sure the count goes up.
b) Insert some text into the log entry that will change each time the test is run. For example pass DateTime.Now.ToString() as the 2nd argument to crashing app, include that string in the message field of the exception and then check for it in the verification.

Reply:
I think either (a) or (b) would have been sufficient on their own, but both is fine too.